### ISSUE TITLE
New version on PyPi not installable with pip

### ISSUE THREAD

**Issue by ViktorHaag** [digitalbazaar/pyld#48]
Using pip 8.1.2 with Python 3.5, PyLD 0.7.1 is not installable (but is installable with Python2):

```
~/src/devsite/client-libraries-dev/python/tests : [master x] pip3.5 install PyLD==0.7.1
Collecting PyLD==0.7.1
Installing collected packages: PyLD
Successfully installed PyLD-0.7.0
~/src/devsite/client-libraries-dev/python/tests : [master x] pip install PyLD==0.7.1 
Collecting PyLD==0.7.1
Installing collected packages: PyLD
Successfully installed PyLD-0.7.1
~/src/devsite/client-libraries-dev/python/tests : [master x] 
```

Note that despite seeing 0.7.1 is there (it doesn't error that no such version exists), it actually collects and installs 0.7.0 for Python3.

(Workaround is to download the *.tar.gz and install with `pip install -e`, but I'm not sure that's a great long term plan.)

**Comment by davidlehn** [comment]
Strange error.  Seems to work for me.  Maybe some cache issue?  Does it work in a fresh virtualenv?

```
/$ mkdir /tmp/venv
/$ cd /tmp/venv
/tmp/venv$ virtualenv -p /usr/bin/python3.5 pyld
Running virtualenv with interpreter /usr/bin/python3.5
Using base prefix '/usr'
New python executable in /tmp/venv/pyld/bin/python3.5
Also creating executable in /tmp/venv/pyld/bin/python
Installing setuptools, pkg_resources, pip, wheel...done.
/tmp/venv$ cd pyld/
/tmp/venv/pyld$ source bin/activate
(pyld) /tmp/venv/pyld$ python --version
Python 3.5.2+
(pyld) /tmp/venv/pyld$ pip --version
pip 8.1.2 from /tmp/venv/pyld/lib/python3.5/site-packages (python 3.5)
(pyld) /tmp/venv/pyld$ pip install PyLD==0.7.1
Collecting PyLD==0.7.1
Installing collected packages: PyLD
Successfully installed PyLD-0.7.1
```

**Comment by ViktorHaag** [comment]
Nope:

```
~/tmp :  virtualenv -p /usr/local/bin/python3.5 pyld
Running virtualenv with interpreter /usr/local/bin/python3.5
Using base prefix '/usr/local/Cellar/python3/3.5.2_1/Frameworks/Python.framework/Versions/3.5'
New python executable in /Users/vhaag/tmp/pyld/bin/python3.5
Also creating executable in /Users/vhaag/tmp/pyld/bin/python
Installing setuptools, pip, wheel...done.
~/tmp :  cd pyld
~/tmp/pyld :  source bin/activate
(pyld) ~/tmp/pyld :  pip freeze
(pyld) ~/tmp/pyld :  python --version
Python 3.5.2
(pyld) ~/tmp/pyld :  pip --version
pip 8.1.2 from /Users/vhaag/tmp/pyld/lib/python3.5/site-packages (python 3.5)
(pyld) ~/tmp/pyld :  pip install PyLD
Collecting PyLD
Installing collected packages: PyLD
Successfully installed PyLD-0.7.0
(pyld) ~/tmp/pyld :  pip uninstall PyLD
Uninstalling PyLD-0.7.0:
  /Users/vhaag/tmp/pyld/lib/python3.5/site-packages/PyLD-0.7.0.dist-info/DESCRIPTION.rst
  /Users/vhaag/tmp/pyld/lib/python3.5/site-packages/PyLD-0.7.0.dist-info/INSTALLER
  /Users/vhaag/tmp/pyld/lib/python3.5/site-packages/PyLD-0.7.0.dist-info/METADATA
  /Users/vhaag/tmp/pyld/lib/python3.5/site-packages/PyLD-0.7.0.dist-info/RECORD
  /Users/vhaag/tmp/pyld/lib/python3.5/site-packages/PyLD-0.7.0.dist-info/WHEEL
  /Users/vhaag/tmp/pyld/lib/python3.5/site-packages/PyLD-0.7.0.dist-info/metadata.json
  /Users/vhaag/tmp/pyld/lib/python3.5/site-packages/PyLD-0.7.0.dist-info/top_level.txt
  /Users/vhaag/tmp/pyld/lib/python3.5/site-packages/pyld/__init__.py
  /Users/vhaag/tmp/pyld/lib/python3.5/site-packages/pyld/__pycache__/__init__.cpython-35.pyc
  /Users/vhaag/tmp/pyld/lib/python3.5/site-packages/pyld/__pycache__/jsonld.cpython-35.pyc
  /Users/vhaag/tmp/pyld/lib/python3.5/site-packages/pyld/jsonld.py
Proceed (y/n)? y
  Successfully uninstalled PyLD-0.7.0
(pyld) ~/tmp/pyld :  pip install PyLD==0.7.1
Collecting PyLD==0.7.1
Installing collected packages: PyLD
Successfully installed PyLD-0.7.0
```

**Comment by davidlehn** [comment]
Try checking output with `pip -vvv install pyld`.  It should list all the versions it finds, and let you know if it's using a cached copy.  if so can add `--no-cache-dir` so it downloads latest.

**Comment by ViktorHaag** [comment]
Yep. That did it. For some reason, it was unable to find a more recent version in its lookup sequence, so it used the pre-built wheel for 0.7.0 and moved on. The wheel was still there, I presume, after I installed with `-e`, hidden, under the source install.

I did a `pip uninstall pyld` twice to clear out the source install and the history of the previous install, and then re-installed with `--no-cache-dir` and it installed fine.

Package managers shouldn't be this tricksy. 8/

Thanks for the assistance!
